sidecar_test.go

```diff
@@ -193,10 +196,105 @@ spec:
     secret:
       secretName: telegraf-config-myname
 status: {}
-`
-	if got != want {
-		t.Errorf("unexpected pod got:\n%v\nwant:\n%v", got, want)
+			`,
+		},
+		{
+			name: "validate custom telegraf image pod definition",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						TelegrafImage: "docker.io/library/telegraf:1.11",
+					},
+				},
+			},
+			wantPod: `
+metadata:
+  annotations:
+    telegraf.influxdata.com/image: docker.io/library/telegraf:1.11
+  creationTimestamp: null
+spec:
+  containers:
+  - env:
+    - name: NODENAME
+      valueFrom:
+        fieldRef:
+          fieldPath: spec.nodeName
+    image: docker.io/library/telegraf:1.11
+    name: telegraf
+    resources:
+      limits:
+        cpu: 500m
+        memory: 500Mi
+      requests:
+        cpu: 50m
+        memory: 50Mi
+    volumeMounts:
+    - mountPath: /etc/telegraf
+      name: telegraf-config
+  volumes:
+  - name: telegraf-config
+    secret:
+      secretName: telegraf-config-myname
+status: {}
+			`,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			secret, err := addSidecar(tt.pod, defaultTelegrafImage, "myname", "mynamespace", "")
+			if err != nil {
+				t.Errorf("unexpected error adding to sidecar: %v", err)
+			}
+
+			if tt.wantSecret != "" {
+				if want, got := strings.TrimSpace(tt.wantSecret), strings.TrimSpace(toYAML(t, secret)); got != want {
+					t.Errorf("unexpected secret got:\n%v\nwant:\n%v", got, want)
+				}
+			}
+
+			if tt.wantPod != "" {
+				if want, got := strings.TrimSpace(tt.wantPod), strings.TrimSpace(toYAML(t, tt.pod)); got != want {
+					fmt.Printf("WTF\n%s\n", got)
+					t.Errorf("unexpected pod got:\n%v\nwant:\n%v", got, want)
+				}
+			}
+		})
 	}
+
```

Comment:
Maybe remove the commented code if not used

Reply:
OMG, I completely forgot about that. Removed.